packages/@expo/cli/src/start/server/__tests__/DevToolsPluginCliExtensionExecutor-test.ts

```diff
@@ -0,0 +1,258 @@
+        },
+      };
+      const args = { param1: `value'with"quotes' && || |##`, param2: 42 };
+      const { close, spawn } = await executePluginCommandAsync({ pluginDescriptor, args });
+      const result = await close(0);
+
+      expect(spawn).toHaveBeenCalledWith(
+        expect.any(String),
+        expect.arrayContaining([expect.stringContaining(JSON.stringify(args))]),
+        expect.any(Object)
+      );
+      expect(result).toEqual([]);
+    });
+
+    it('should handle errors thrown by the command', async () => {
+      const pluginDescriptor = {
+        ...PLUGIN_DESCRIPTOR,
+        cliExtensions: {
+          commands: [COMMAND],
+          description: 'Test Plugin',
+          entryPoint: 'index.js',
+        },
+      };
+
+      const { close } = await executePluginCommandAsync({ pluginDescriptor });
+      const result = await close(1);
+
+      expect(result).toEqual([
+        { type: 'text', level: 'error', text: 'Process exited with code 1' },
+      ]);
+    });
+  });
+
+  describe('getCommandString', () => {
+    it('should return the command string for a command without parameters', () => {
+      const pluginDescriptor = {
+        ...PLUGIN_DESCRIPTOR,
+        cliExtensions: {
+          commands: [COMMAND],
+          description: 'Test Plugin',
+          entryPoint: 'index.js',
+        },
+      };
+      const executor = new DevToolsPluginCliExtensionExecutor(pluginDescriptor, PROJECT_ROOT);
+      const commandString = executor.getCommandString({
+        command: 'test-command',
+        args: {},
+      });
+      expect(commandString).toBe(`node index.js test-command`);
+    });
+
+    it('should return the command string for a command with parameters', () => {
+      const pluginDescriptor = {
+        ...PLUGIN_DESCRIPTOR,
+        cliExtensions: {
+          commands: [COMMAND_WITH_PARAMS],
+          description: 'Test Plugin',
+          entryPoint: 'index.js',
+        },
+      };
+      const executor = new DevToolsPluginCliExtensionExecutor(pluginDescriptor, PROJECT_ROOT);
+      const commandString = executor.getCommandString({
+        command: 'test-command',
+        args: { param1: 'value1', param2: 42 },
+      });
+      expect(commandString).toBe(`node index.js test-command {"param1":"value1","param2":42}`);
+    });
+  });
+});
+
+// --------------- HELPERS  ---------------
+
+const executePluginCommandAsync = async (params: {
+  pluginDescriptor: DevToolsPluginInfo;
+  args?: Record<string, any>;
+  timeoutMs?: number;
+  spawnFunc?: typeof import('child_process').spawn;
+}) => {
+  const { pluginDescriptor, args = {}, timeoutMs = 10_000, spawnFunc } = params;
+  const closeListeneres: ((exitCode: number) => void)[] = [];
+  let resolver: (() => void) | null = null;
+  const closePromise = new Promise<void>((resolve) => (resolver = resolve));
+  let log = '';
+  let err = '';
+  const kill = jest.fn();
+  const mock = {
+    spawn:
+      spawnFunc ||
```

Comment:
```suggestion
      spawnFunc ??
```

because it's a function, maybe use `??` makes more sense